libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -1168,9 +1187,45 @@ void AP_CRSF_Telem::calc_parameter() {
         _telem.ext.param_entry.payload[idx++] = 't';
         _telem.ext.param_entry.payload[idx++] = 0; // null terminator
 
+#if AP_CRSF_SCRIPTING
+        if (scripted_menus.next_menu != nullptr) {
+            _telem.ext.param_entry.payload[idx++] = PARAMETER_MENU_ID; // root parameter screen ardupilot menu
+            for (ScriptedMenu* m = scripted_menus.next_menu; m != nullptr; m = m->next_menu) {
+                if (m->parent_id == 0) {
+                    _telem.ext.param_entry.payload[idx++] = m->id;   // scripted menus
+                }
+            }
+        } else
+#endif // AP_CRSF_SCRIPTING
+        {
+            // write out all of the ids we are going to send
+            for (uint8_t i = 0; i < AP_OSD_ParamScreen::NUM_PARAMS * AP_OSD_NUM_PARAM_SCREENS; i++) {
+                _telem.ext.param_entry.payload[idx++] = PARAMETER_MENU_ID + i + 1;
+            }
+        }
+        _telem.ext.param_entry.payload[idx] = 0xFF; // terminator
+
+        _telem_size = sizeof(AP_CRSF_Telem::ParameterSettingsEntryHeader) + 1 + idx;
+        _telem_type = AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAMETER_SETTINGS_ENTRY;
+        _pending_request.frame_type = 0;
+        _telem_pending = true;
+        return;
+    }
+
+#if AP_CRSF_SCRIPTING
+    // root folder request
+    if (scripted_menus.next_menu != nullptr && _param_request.param_num == PARAMETER_MENU_ID) {
+        _telem.ext.param_entry.header.param_num = PARAMETER_MENU_ID;
+        _telem.ext.param_entry.header.chunks_left = 0;
+        _telem.ext.param_entry.payload[idx++] = 0; // parent folder
+        _telem.ext.param_entry.payload[idx++] = ParameterType::FOLDER; // type
+        // name
+        strncpy((char*)&_telem.ext.param_entry.payload[idx], "Parameters", 16);
```

Comment:
```suggestion
        strncpy((char*)&_telem.ext.param_entry.payload[idx], "Parameters", ARRAY_SIZE(_telem.ext.param_entry.payload) - idx);
```

... and I'm otherwise concerned with indexing into the payload here without checking that idx isn't walking off the end of the array.  We'd want a really big comment in here explaining how we don't overwrite this buffer...

Reply:
Fixed thanks

Reply:
Doesn't work:
```
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘void AP_CRSF_Telem::calc_parameter()’ at ../../libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:1252:16:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:34: error: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ specified bound 54 equals destination size [-Werror=stringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   96 |                                   __glibc_objsize (__dest));
```